tests/Unit/DataStructures/Test_DataVector.cpp

```diff
@@ -173,53 +173,45 @@ void check_vectors(const T1& t1, const T2& t2) {
 
 SPECTRE_TEST_CASE("Unit.DataStructures.DataVector.MathAfterMove",
                   "[Unit][DataStructures]") {
-  DataVector m0(10, 3.0), m1(10, 9.0);
+  const DataVector m0(10, 3.0);
+  const DataVector m1(10, 9.0);
+  // Test math after a move assignment
   {
     DataVector a(10, 2.0);
     DataVector b{};
     b = std::move(a);
-    b = m0 + m1;
-    check_vectors(b, DataVector(10, 12.0));
     // clang-tidy: use after move (intentional here)
     CHECK(a.size() == 0);  // NOLINT
     CHECK(a.is_owning());
+    CHECK(b.size() == 10);
+    CHECK(b.is_owning());
+    check_vectors(b, DataVector(10, 2.0));
+    b = m0 + m1;
+    CHECK(a.size() == 0);  // NOLINT
+    check_vectors(b, DataVector(10, 12.0));
     a = m0 * m1;
     check_vectors(a, DataVector(10, 27.0));
     check_vectors(b, DataVector(10, 12.0));
   }
 
-  {
-    DataVector a(10, 2.0);
```

Comment:
Please revert these changes.  These tests were not redundant, but were testing for a regression of a bug involving moving erroneously causing a partial aliasing of DataVectors.  In particular, the move from `a` to `b` followed by an expression template assignment to `a` with no intervening modification of `b` is important to the test.